paraatm/HPMM/hpmm.py

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/python3 python3
+# -*- coding: utf-8 -*-
+
+"""
+@Author: Jiawei Chen
+@Date: 2020-05-20
+This Python script is used for HPMM (Human Performance Monitoring Module)
+The script outputs predicted loss of separation as an indicator of air traffic controller's performance
+The script can also check compliance for each command
+@Last Modified by: Jiawei Chen
+@Last Modified date: 2020-05-26
+"""
+from typing import Any, Union
+
+from jpype import *
+from array import *
+import pandas as pd
+import numpy as np
+import scipy.stats as sts
+import os
+import time
+import sys
+import datetime
+from shutil import copyfile
+from shutil import rmtree
+
+import matplotlib.pyplot as plt
+from pandas import Series, DataFrame
+from pandas.io.parsers import TextFileReader
+from scipy.stats import norm
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+import catboost as cb
+from catboost import CatBoostRegressor
+from catboost import Pool
+
+
+
+class HPMM(object):
+    def __init__(self, cfg):
+        self.data = pd.read_csv(cfg['datafile']);
+
+    def preprocessing(self):
+        # according to experiment design, remove redundant variables
+        data = self.data
+        data_domain = data.drop(
+            columns=['Ss', 'condtn', 'los_dur_over5min', 'query_timed_out', 'ready_timed_out', 'ready_latency_adj',
+                     'cum_los_dur',
+                     'stimuli', 'response_text', 'condtn_num'])
+        # fill in missing values
+        data_domain['sa_correct'].fillna(data_domain['sa_correct'].mode()[0], inplace=True)
+        data_domain.fillna(data_domain.mean(), inplace=True)
+
+        # transform categorical data
+        data_domain = data_domain.join(pd.get_dummies(data_domain.sa_correct))
+        data_domain = data_domain.join(pd.get_dummies(data_domain['query']))
+        data_domain = data_domain.drop(columns=['query', 'sa_correct'])
+        return data_domain
+
+    def modelpredict(model, input):
+        return model.predict(input)
+
+    def train(self, model, train_pool, test_pool, plot):
+        model.fit(train_pool, eval_set=test_pool, plot=plot)
+
+
+if __name__ == '__main__':
+    # my code
+    cfg = {
+        'datafile': '/home/jchen311/para-atm/paraatm/sample_data/human_data.csv'
+    }
```

Comment:
Please remove this reference to your local file system.  This will not work on anyone's system but yours.  See [test_para_atm.py](../blob/master/paraatm/tests/test_para_atm.py) for examples of the best practice for locating files in the `sample_data` directory in a universal way.